api/types/events/api.go

```diff
@@ -69,6 +69,12 @@ type AuditEvent interface {
 	GetClusterName() string
 	// SetClusterName sets the name of the teleport cluster on the event.
 	SetClusterName(string)
+
+	// TrimToMaxSize returns a copy of the event trimmed to a smaller
+	// size. The desired size may not be achievable so callers
+	// should always check the size of the returned event before
+	// using it.
+	TrimToMaxSize(maxSize int) AuditEvent
```

Comment:
Does `TrimToMaxSize` cap on maximum or minimum size for the `maxSize` input? What are the chances of providing a smaller maxSize than necessary and loosing out important event data? Is that something we need to worry about?

Reply:
There's no hard min or max cap that is known beforehand, because only certain fields are trimmed the minimum `maxSize` input really depends on what other fields are set to in the event (ie in `UserMetadata` etc). I don't think it's really something to worry about, all callers of  `TrimToMaxSize` check that the trimmed event is small enough so this shouldn't be an issue